api/app.js

```diff
@@ -2,27 +2,31 @@ import express from 'express';
 import cors from 'cors';
 import { healthCheckRouter } from './Routes/healthRoute.js';
 import { testPermissionRouter } from './Routes/testPermissionsRoute.js';
-const PORT = 8080;
+import { authenticationMiddleware } from './Middleware/authenticationMiddleware.js';
+
+const port = process.env.PORT || 8080;
+
 export const app = express();
 
-const frontend=process.env.FRONT_END_URL || 'http://localhost:5500';
+const frontendUrl = process.env.FRONT_END_URL || 'http://localhost:5500';
 const corsOptions = {
-    'origin': frontend,
-    'methods': 'GET,HEAD,PUT,PATCH,POST,DELETE',
-    'preflightContinue': false,
-    'optionsSuccessStatus': 204
+    origin: frontendUrl,
+    methods: 'GET,HEAD,PUT,PATCH,POST,DELETE',
+    preflightContinue: false,
+    optionsSuccessStatus: 204,
 };
 
 app.use(cors(corsOptions));
+app.use(authenticationMiddleware);
```

Comment:
Maybe only add this to the routes that need authentication, basically everything but health check